packages/repository-json-schema/src/__tests__/integration/build-schema.integration.ts

```diff
@@ -1082,5 +1082,20 @@ describe('build-schema', () => {
         expect(optionalNameSchema.title).to.equal('ProductPartial');
       });
     });
+
+    it('does not attempt to assign to undefined when building schema for recursive entities', () => {
+      @model()
+      class Category extends Entity {
+        @property({id: true})
+        id: number;
+
+        @hasMany(() => Category)
+        subCategories?: Category[];
+      }
+
+      expect(getJsonSchema(Category, {includeRelations: true})).not.throwError(
+        "Cannot set property 'CategoryWithRelations' of undefined",
```

Comment:
Let's improve this test to check for the desired outcome (how should the JSON schema look like) instead of merely checking that a specific error was not thrown.

```ts
const schema = await getJsonSchema(Category, {includeRelations: true});
expect(schema).to.deepEqual({
  // what should the schema look like?
});
```

Maybe we don't need to assert on the entire schema object, just check how `subCategories` property is defined and how does it reference the root schema.